pss/pss_test.go

```diff
@@ -1777,6 +1794,65 @@ func benchmarkSymkeyBruteforceSameaddr(b *testing.B) {
 	}
 }
 
+func BenchmarkMessageProcessing(b *testing.B) {
+	b.Run(fmt.Sprintf("FailProb%.2f", 0.0), func(b *testing.B) { benchmarkMessageProcessing(b, 0.0) })
+	b.Run(fmt.Sprintf("FailProb%.2f", 0.01), func(b *testing.B) { benchmarkMessageProcessing(b, 0.01) })
+	b.Run(fmt.Sprintf("FailProb%.2f", 0.05), func(b *testing.B) { benchmarkMessageProcessing(b, 0.05) })
+}
+
+func benchmarkMessageProcessing(b *testing.B, failProb float32) {
+	rand.Seed(0)
+	// setup
+	privkey, _ := crypto.GenerateKey()
+
+	addr := make([]byte, 32)
+	addr[0] = 0x01
+	ps := newTestPssStart(privkey, network.NewKademlia(addr, network.NewKadParams()), NewParams(), false)
+
+	numMessages := 200000
+	ps.outbox = newOutbox(numMessages)
+	failed := 0
+	procChan := make(chan struct{}, numMessages)
+	succesForward := func(msg *PssMsg) bool {
+		roll := rand.Float32()
+		if failProb > roll {
+			failed++
```

Comment:
race on incrementing

Reply:
I will remove the failed counter as we are not using it anymore.

Reply:
Removed counter in commit dad34db